components/AddressInput.test.tsx

```diff
@@ -2,26 +2,33 @@ import React from 'react'
 import { render, fireEvent, act } from '@testing-library/react'
 import { waitFor } from '@testing-library/dom'
 import AddressInput from './AddressInput'
-import assert from 'assert'
+import * as nextRouter from 'next/router'
+
+// @ts-expect-error mocked next router
+nextRouter.useRouter = jest.fn()
+
+// @ts-expect-error mocked next router
+nextRouter.useRouter.mockImplementation(() => ({ route: '/' }))
 
 describe('AddressInput', () => {
   it('renders no initial value', () => {
+    let input: HTMLInputElement | null
     act(() => {
       const { container } = render(<AddressInput />)
-      expect(container.querySelector('input')).toHaveAttribute('value', '')
+      input = container.querySelector('input')
     })
+    waitFor(() => expect(input).toHaveAttribute('value', ''))
   })
 
   it('renders initial value', () => {
+    let input: HTMLInputElement | null
     act(() => {
       const { container } = render(
         <AddressInput recipientWalletAddress={'0xfoo.eth'} />
       )
-      expect(container.querySelector('input')).toHaveAttribute(
```

Comment:
Is this change required by the change to React 18, or did this just sneak in?

Reply:
Not very confident, but without waitFor it was failing and giving null reference for the loaded dom, I tweaked this therefore, maybe a react 18 thing, because on other PR's all of this worked